### ISSUE TITLE
Move some helper methods from tester to find

### ISSUE THREAD

**Issue by HerrNiklasRaab** [flutter/flutter#96170]
@Hixie In [this PR](https://github.com/flutter/flutter/pull/3622) all the following methods have been moved to the `WidgetController` class:

any() - true if anything matches, c.f. Iterable.any
allWidgets - all the widgets in the tree
widget() - the one and only widget that matches the finder
firstWidget() - the first widget that matches the finder
allElements - all the elements in the tree
element() - the one and only element that matches the finder
firstElement() - the first element that matches the finder
allStates - all the States in the tree
state() - the one and only state that matches the finder
firstState() - the first state that matches the finder
allRenderObjects - all the render objects in the tree
renderObject() - the one and only render object that matches the finder
firstRenderObject() - the first render object that matches the finder

I am curious why this decision was made, because none of the methods uses members from the `WidgetController` class.  Wouldn't have made it more sense to move those methods to `find`?

Instead of:
`tester.widget(find.text("Hallo"));`

You could have written the shorter and easier to read:
`find.text("Hallo").widget;`

**Comment by maheshmnj** [comment]
Keeping this issue open for further insights from the team.

**Comment by HerrNiklasRaab** [comment]
Adding to this: I wasn't even aware that those methods even exist (until today), neither did my colleague, both of us were Full-Time Flutter programmers for the last 2 years. So we were quite surprised to see this.

**Comment by Hixie** [comment]
I agree that the API is not really sensible. I've come to the conclusion that we should probably have designed the test system very differently (and not used package:test at all), but unfortunately there's so many tests that rely on these APIs now that it's going to be really difficult to migrate to a new model (and having two models is even worse).

**Comment by HerrNiklasRaab** [comment]
To be honest, I would very much appreciate it if the Flutter Framework always tries to keept it's best version, even if it means that people need to make some adjustments in their code. If we are afraid of making such changes, we are making long-term sacrifices for short-term gains.

Can't we just deprecate the methods? This would break nobodies code, but lead to the correct design for all the new code written. I can make a PR!

**Comment by Hixie** [comment]
We can try. Let's begin with a design doc that describes the whole API, so that we can evaluate the whole thing in context, and compare it to what we have today and see how much of an improvement it is? See https://github.com/flutter/flutter/wiki/Chat#design-documents
